### PR TITLE
Refactoring/azure/storageaccounts

### PR DESCRIPTION
**Pull request by misg**
This PR aims to migrate Azure Storage Accounts service to the new architecture, as part of #183.

### CODE REVIEW

File: ScoutSuite/providers/azure/resources/storageaccounts/blob_containers.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    async def fetch_all(self):
+        raw_blob_containers = await self.facade.get_blob_containers(
+            self.resource_group_name, self.storage_account_name
+        )
+        for raw_blob_container in raw_blob_containers:
+            id, blob_container = self._parse(raw_blob_container)
+            self[id] = blob_container
+
+    def _parse(self, raw_blob_container):
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/storageaccounts/blob_containers.py]:
```suggestion
    def _parse_blob_container(self, raw_blob_container):
```

File: ScoutSuite/providers/azure/resources/storageaccounts/blob_containers.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        self.resource_group_name = resource_group_name
+        self.storage_account_name = storage_account_name
+        self.facade = facade
+
+    async def fetch_all(self):
+        raw_blob_containers = await self.facade.get_blob_containers(
+            self.resource_group_name, self.storage_account_name
+        )
+        for raw_blob_container in raw_blob_containers:
+            id, blob_container = self._parse(raw_blob_container)
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/storageaccounts/blob_containers.py]:
```suggestion
            id, blob_container = self._parse_blob_container(raw_blob_container)
```

File: ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    storage_account_name=storage_account['name'],
+                    facade=facade
+                )
+            ) for storage_account in self['storage_accounts'].values()
+        }
+        await asyncio.wait(tasks)
+
+        self['storage_accounts_count'] = len(self['storage_accounts'])
+
+    def _parse(self, raw_storage_account):
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py]:
```suggestion
    def _parse_storage_account(self, raw_storage_account):
```

File: ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        (BlobContainers, 'blob_containers')
+    ]
+
+    async def fetch_all(self, credentials, **kwargs):
+        # TODO: build that facade somewhere else:
+        facade = StorageAccountsFacade(credentials.credentials, credentials.subscription_id)
+
+        self['storage_accounts'] = {}
+        for raw_storage_account in await facade.get_storage_accounts():
+            id, storage_account = self._parse(raw_storage_account)
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py]:
```suggestion
            id, storage_account = self._parse_storage_account(raw_storage_account)
```

File: ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    _children = [
+        (BlobContainers, 'blob_containers')
+    ]
+
+    async def fetch_all(self, credentials, **kwargs):
+        # TODO: build that facade somewhere else:
+        facade = StorageAccountsFacade(credentials.credentials, credentials.subscription_id)
+
+        self['storage_accounts'] = {}
+        for raw_storage_account in await facade.get_storage_accounts():
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py]: I feel like having this kind of breaks the convention. It's the same as if you had a child named `storage_accounts`, basically.

Comment (Aboisier) [reply ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py]: Ok, I see why you did this. Still breaks the structure. Is there a way we can reorganize this so we don't break the conventions?

Comment (misg) [reply ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py]: It's not the only place where I'm doing that unfortunately... It's strictly for compatibility reasons and I was planning a PR to get rid of that as soon as we will be moving to the new architecture.

Comment (Aboisier) [reply ScoutSuite/providers/azure/resources/storageaccounts/storageaccounts.py]: Sounds good!

**Pull request merged**